baseline/ica/data_processing.py

```diff
@@ -53,48 +52,51 @@ def try_load_composition_df(self, composition_data_loc: str) -> bool:
 
     def preprocess(self, calib_data_path: Path, average_locations=False, norm: Norm = Norm.NORM_1) -> None:
         sample_data = get_preprocessed_sample_data(self.sample_name, calib_data_path, average_shots=False)
-        location_name_ss, single_sample = list(sample_data.items())[0]
 
-        self.sample_id = self.sample_name if average_locations else f"{self.sample_name}_{location_name_ss}"
+        dfs = []
 
-        # Average all of the five location datasets into one single dataset
-        final_avg_shots_df = (
-            average_each_shot_across_locations(sample_data) if average_locations else single_sample
-        )
+        if average_locations:
+            dfs.append((self.sample_name, average_each_shot_across_locations(sample_data)))
+        else:
+            for location_name, location_df in sample_data.items():
+                dfs.append((location_name, location_df))
```

Comment:
You could probably just replace this with a list comprehension (which is usually faster than iteration) and a ternary operator (to avoid defining dfs as an empty list first before filling it in). This in-place initialization should be faster and is also more concise (and, dare I say, more Pythonic)

```python
        dfs = [(self.sample_name, average_each_shot_across_locations(sample_data))] if average_locations else [(location_name, location_df) for location_name, location_df in sample_data.items()]
```